src/Controller/HearingController.php

```diff
@@ -369,13 +370,45 @@ public function hearingPostRequestEdit(CaseEntity $case, DocumentRepository $doc
     /**
      * @Route("/{case}/hearing/{hearingPost}/approve", name="case_hearing_post_approve", methods={"POST"})
      */
-    public function hearingPostApprove(CaseEntity $case, HearingPostResponse $hearingPost): Response
+    public function hearingPostApprove(CaseEntity $case, HearingPostResponse $hearingPost, MailTemplateHelper $mailTemplateHelper, DocumentUploader $documentUploader, DigitalPostHelper $digitalPostHelper): Response
     {
         $this->denyAccessUnlessGranted('edit', $case);
 
         if ($hearingPost->getHearing()->getFinishedOn()) {
             throw new HearingException();
         }
+
+        // Send receipt
+        if ($hearingPost->getSendReceipt()) {
+            $sender = $hearingPost->getSender();
+            $digitalPostRecipients[] = (new DigitalPost\Recipient())
+                ->setName($sender->getName())
+                ->setIdentifierType($sender->getIdentification()->getType())
+                ->setIdentifier($sender->getIdentification()->getIdentifier())
+                ->setAddress($sender->getAddress())
+            ;
+            $case = $hearingPost->getHearing()?->getCaseEntity();
+            $template = $case?->getBoard()?->getReceiptHearingPost();
```

Comment:
`$case` and `$template` may be `null` so you should check for this before using the values in the code below.

Reply:
Added check.